Fix artifact file permissions (#702, #853)

shoriminimoe opened the pull request:
Sets artifact file permissions to octal `600` in
`ansible_runner.utils.dump_artifact`

Fixes: #702, #853

shoriminimoe committed:
Fix artifact file permissions (#702, #853)

Sets artifact file permissions to octal `600` in
`ansible_runner.utils.dump_artifact`

Akasurde commented:
@shoriminimoe Thanks for the PR. Could you please add an integration test for this change?

shoriminimoe commented:
Will do.

shoriminimoe committed:
test: add file permission tests

Adds `test_artifact_permissions` unit test and
`test_repeat_run_with_new_inventory` integration test.
`test_artifact_permissions` verifies that artifacts are created with
user read/write permission. `test_repeat_run_with_new_inventory`
verifies repeatedly running a playbook with a different inventory
hostname is successful.

shoriminimoe commented:
@Akasurde, tests are added in https://github.com/ansible/ansible-runner/pull/1183/commits/9bbf512a941210c892747b869ad03f26b4794f6f

The unit test verifies `dump_artifact` creates artifacts with _at least_ user read/write permission.

The integration test runs a playbook twice with different inventory host values. Doing so should always succeed. This method is similar to what others have encountered in #702 and #853. 

I have verified that both tests fail before https://github.com/ansible/ansible-runner/pull/1183/commits/85b5d7cdc3fdbeba06ae2d152851813908b970a7 and are successful after including it.

Shrews reviewed (approved):
@shoriminimoe Thanks for your contribution.

Akasurde commented:
@shoriminimoe Thanks for the contribution.
